hub/api/dataset.py

```diff
@@ -89,43 +81,50 @@ def __getitem__(self, item: Union[str, int, slice, Index]):
         else:
             raise InvalidKeyTypeError(item)
 
-    def __setitem__(self, item: Union[slice, str], value):
-        if isinstance(item, str):
-            tensor_key = item
+    def create_tensor(
+        self, name: str, chunk_size: int = DEFAULT_CHUNK_SIZE, dtype: str = "float64"
+    ):
+        """Create a new tensor in this dataset.
 
-            if tensor_exists(tensor_key, self.provider):
-                raise TensorAlreadyExistsError(tensor_key)
+        Args:
+            name (str): The name of the tensor to be created.
+            chunk_size (int): The target size for chunks in this tensor.
+            dtype (str): The dtype to use for this tensor.
+                Will be overwritten when the first sample is added.
 
-            if isinstance(value, np.ndarray):
-                tensor_meta = tensor_meta_from_array(value, batched=True)
+        Returns:
+            The new tensor, which can also be accessed by `self[name]`.
 
-                ds_meta = read_dataset_meta(self.provider)
-                ds_meta["tensors"].append(tensor_key)
-                write_dataset_meta(self.provider, ds_meta)
+        Raises:
+            TensorAlreadyExistsError: Duplicate tensors are not allowed.
```

Comment:
Can we check if tensor already exists and raise an error only if the new arguments for tensor creation don't match with the old one?

Reply:
@AbhinavTuli why? I feel like duplicate `create` calls are still invalid. How do you feel about a `strict=True` parameter to determine if we should throw in that case?